proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+    recipientAddress = ccm.params.recipientAddress
+    data = ccm.params.data
+    sendingChainID = ccm.sendingChainID
+
+    if getChainID(nftID) not in [OWN_CHAIN_ID, sendingChainID]:
+        raise Exception("NFT is not native to either the sending chain or the receiving chain.")
+```
+
 ##### Execution
 
 When executing a cross-chain NFT transfer message `ccm`, the logic below is followed.
 
 ```python
-nftID = ccm.params.nftID
-chainID = nftID.chainID
-sendingChainID = ccm.sendingChainID
-senderAddress = ccm.params.senderAddress
-recipientAddress = ccm.params.recipientAddress
-receivedAttributes = ccm.params.attributes
-ownChainID = interoperability.getOwnChainAccount().ID
-
-if (chainID not in [ownChainID, sendingChainID]
-    or length(senderAddress) != ADDRESS_LENGTH
-    or length(recipientAddress) !=  ADDRESS_LENGTH
-    or length(receivedAttributes) > MAX_BYTE_LENGTH_ATTRIBUTES
-    or (chainID == ownChainID
-        and NFTStore(nftID).owner != uint32(sendingChainID))
-    or (chainID == sendingChainID
-        and NFTStore(nftID) exists)):  
-    if  (ccm.fee >= MIN_RETURN_FEE * length(ccm)
-        and ccm.status == CCM_STATUS_OK):
-        interoperability.error(ccm, CCM_STATUS_NFT_PROTOCOL_VIOLATION)
-    interoperability.terminateChain(sendingChainID)
-    stop ccm execution
-
-if chainID == ownChainID:
-    oldAttributes = NFTStore(nftID).attributes
-    if ccm.status == CCM_STATUS_OK:
-        newAttributes = getNewAttributes(nftID.collection,
-                                         oldAttributes,
-                                         receivedAttributes)
-        newRecipientAddress = recipientAddress
-    else:
-        newAttributes = oldAttributes
-        newRecipientAddress = senderAddress
-
-    nftID = canonicalNFTID(nftID)
-    NFTStore(nftID).owner = newRecipienAddress
-    NFTStore(nftID).attributes = newAttributes
-    createUserEntry(newRecipientAddress, nftID)
-
-else: # chainID == sendingChainID
-    if NFTSupported(nftID) == FALSE:
-        if (ccm.fee >= MIN_RETURN_FEE * length(ccm)
-            and ccm.status == CCM_STATUS_OK):
-            interoperability.error(ccm, CCM_STATUS_NFT_NOT_SUPPORTED)
-        stop ccm execution
-    if ccm.status == CCM_STATUS_OK:
-        createNFTEntry(nftID,
-                       recipientAddress,
-                       receivedAttributes)
+def execute(
+    ccu: Transaction,
+    ccm: CCM
+) -> None:
+
+    nftID = ccm.params.nftID
+    senderAddress = ccm.params.senderAddress
+    recipientAddress = ccm.params.recipientAddress
+    data = ccm.params.data
+    sendingChainID = ccm.sendingChainID
+
+    if isNFTSupported(nftID) == False:
+        emitPersistentEvent(
+            module = MODULE_NAME_NFT,
+            name = EVENT_NAME_CCM_TRANSFER,
+            data = {
+                "senderAddress": senderAddress,
+                "recipientAddress": recipientAddress,
+                "nftID": nftID,
+                "receivingChainID": OWN_CHAIN_ID,
```

Comment:
Why do we add the receivingChainID to the data and, say, not the sendingChainID? I would consistently add both orr neither (maybe neither is better?) Notice that bot values can be recovered from the ccm anyway (and the ccm ID is by default one of the event topics)